From: Oleg Nesterov <oleg@redhat.com>
Date: Mon, 7 Jun 2010 21:41:08 -0400
Subject: [misc] workqueue: make cancel_work_sync EXPORT_SYMBOL_GPL
Message-id: <20100607214108.GA20057@redhat.com>
Patchwork-id: 26017
O-Subject: [RHEL5 PATCH 9/7] bz#596626:  EXPORT_SYMBOL(cancel_work_sync)
	needs _GPL
Bugzilla: 596626
RH-Acked-by: Stanislaw Gruszka <sgruszka@redhat.com>
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

cancel_work_sync is upstream as EXPORT_SYMBOL_GPL. We need to be
_GPL here too,

Suggested-by: Jarod Wilson <jarod@redhat.com>
Signed-off-by: Oleg Nesterov <oleg@redhat.com>

diff --git a/kernel/workqueue.c b/kernel/workqueue.c
index 58d5669..a115360 100644
--- a/kernel/workqueue.c
+++ b/kernel/workqueue.c
@@ -477,7 +477,7 @@ int cancel_work_sync(struct work_struct *work)
 	work->pending = 0; /* clears both "pending" and "valid" bits */
 	return ret;
 }
-EXPORT_SYMBOL(cancel_work_sync);
+EXPORT_SYMBOL_GPL(cancel_work_sync);
 
 static inline void cwq_basic_init(struct workqueue_struct *wq, int cpu)
 {
